### ISSUE TITLE
Ads shows per location

### ISSUE THREAD

**Issue by tsbmedia** [WWBN/AVideo#9707]
Hi guys,
1.  Is it possible that the advertiser's Advert to appear to users of chosen locations ? Example Advertiser from Europe want his ads to appear on Europe continent users unlike to Africa since his targeted audience is European users .

Secondly is there a way where same advertiser can monitor his ads campaign performance . Example see how many times has advert been shown / how many has seen the ad? maybe cpm or cpc been used or even CPM

**Comment by DanielnetoDotCom** [comment]
what exactly type of ads are you referring to?

**Comment by tsbmedia** [comment]
![image](https://github.com/user-attachments/assets/62a6deec-5557-4f6a-af1d-17fe4509c998)

I created that lingerie Ad so my question is it possible that it appear on users of certain location example users in Africa. My second question was can such advert be on CPM cost per mile ? For it to be so what is needed ?

**Comment by DanielnetoDotCom** [comment]
So I guess you are using the ADs plugin

Sorry we do not have a location control per ad, and also we do not control the CPM for those ads

the only ad we do have a location control is the AD_Server https://github.com/WWBN/AVideo/wiki/AD_Server-Plugin

**Comment by tsbmedia** [comment]
Yes Ads plugin .No problem i shall set rates on the per day fixed rate. Thank you for response. i created an ad campaign, uploaded advert video and set parameters to pre roll but when i test the advert on video it does not play at all . Is there something i am missing ?
![image](https://github.com/user-attachments/assets/99d847c5-6d0e-4eaa-987e-d9889b9ef509)

![image](https://github.com/user-attachments/assets/2f6d3e09-7a97-437c-aaee-873f1ea98c7d)
![image](https://github.com/user-attachments/assets/caada924-f923-4de7-b7bb-2e3539cbf653)
![image](https://github.com/user-attachments/assets/5bf3e809-97c8-4a30-9569-3e7539928f9d)

![image](https://github.com/user-attachments/assets/0e4f3ed2-0bb4-40f1-8d07-b620723f7d9f)

**Comment by DanielnetoDotCom** [comment]
Try an incognito browser.

Ads will not show to admin.

**Comment by tsbmedia** [comment]
Meaning ia should logo out and test as user ?

**Comment by DanielnetoDotCom** [comment]
Ingonito browser should work

**Comment by tsbmedia** [comment]
i logged out default ads are showing but the one i created still not showing

**Comment by DanielnetoDotCom** [comment]
Make sure you are not using GoogleAds and ADServer together.

**Comment by tsbmedia** [comment]
okay let me check . Google Ads off but still same issue created ad not showing

**Comment by tsbmedia** [comment]
https://demo.avideo.com/video/447 i tried on this video too doesn't play even video takes too long to load

**Comment by DanielnetoDotCom** [comment]
This video is an embeded youtube video, to play ads a self hosted video is recommende. 

try this one https://demo.avideo.com/video/436/mp4-test-on-mobile

**Comment by tsbmedia** [comment]
Its playing on the link you just send me. Does that mean ads wont play on embedded videos ? only play on direct uploaded ones ?

**Comment by DanielnetoDotCom** [comment]
yes, it is not made to be supported on embed

**Comment by tsbmedia** [comment]
Thanks a lot sir.

**Comment by tsbmedia** [comment]
Is it possible to modify ad sever plugin to work as follows :  
1. Advertiser be the one to create an advert . Ads be on CPM Cost Per Mile and Cost per click CPC which means fee will differ (Admin be able to set the rate )
2. Preview his Ad ?
3. Pay for it ? 
5. Both Admin and advertiser be able to view advert performance example how many views had advert reached . I checked on current Ads sever only Admin can see advert report but what about advertiser ? 
Admin just review the Ad if it does comply with platform standard example its not nudes
Then approve such advert then it goes live .

Sample below 
![image](https://github.com/user-attachments/assets/74a89dbb-efb7-4eba-831d-54c156c6657d)
![image](https://github.com/user-attachments/assets/07282181-248a-4600-b3e9-6b50aed41567)
![image](https://github.com/user-attachments/assets/c614d5bc-0458-47c8-9062-b54dd64b1b1a)
![image](https://github.com/user-attachments/assets/a84a9f2e-fd41-4f91-825f-1a94c1d1193c)

i think by so doing Admin will have lesser task for working on adverts placements and on other hand Advertiser will be able to buy campaigns based on own budget and also be able to view own campaign performances instead of requesting Admin for reports . 

Lastly i am in South Africa and would like to use our local payment gateway let me know if it can be integrated or not and costings thereof   https://developers.payfast.co.za/docs?_gl=1%2a1qxaw54%2a_gcl_aw%2aR0NMLjE3MzU1NDgzNTcuQ2p3S0NBaUFwc203QmhCWkVpd0F2SXUyWDNRWWpnT1NtQ1BOV3AzNURBNks5OVVqb2J1RTl4eW1ua0E3WEkzczhyaWI5U3hpTUNUd2F4b0NNTUVRQXZEX0J3RQ..%2a_gcl_au%2aMjgxNDE4NTAyLjE3MzU1NDgzNTI.%2a_ga%2aMTYzMzY5ODgzNC4xNzM1NTQ4MzU3%2a_ga_LDV3ZJ8YHP%2aMTczNTU0ODM1Ni4xLjAuMTczNTU0ODM1Ni42MC4wLjA.#api-integration

Thank you guys.

**Comment by tsbmedia** [comment]
Images are from other platform i had , its just steps Advertiser does to place advert. On admin site we just see all ads paid ? review them and activate them to go live and they will go live based on dates Advertiser choose and end when CPM end / prints is finished or end date whichever comes first

**Comment by DanielnetoDotCom** [comment]
The ideas you’ve outlined are excellent. While I do see great potential in implementing these features there is no ETA for them at this moment.  

To prioritize these enhancements and establish a timeline, sponsorship would be required to cover the development effort. If you’re interested in sponsoring these features or would like more details, feel free to reach out via email at [developer@youphptube.com](mailto:developer@youphptube.com).

Thank you for your understanding and support!

**Comment by tsbmedia** [comment]
Hi Daniel I am making follow up on the last post you just replied
concerning Ads sever

**Comment by tsbmedia** [comment]
As I posted on github lately
